CRM/Utils/Check/Component/Cms.php

```diff
@@ -16,6 +16,45 @@
  */
 class CRM_Utils_Check_Component_Cms extends CRM_Utils_Check_Component {
 
+  /**
+   * For sites running from Composer, ensure public assets have been correctly installed.
+   *
+   * @return CRM_Utils_Check_Message[]
+   */
+  public static function checkPublicAssets(): array {
+    $checks = [];
+    $assetDir = $GLOBALS['civicrm_asset_map']['civicrm/civicrm-core']['dest'] ?? NULL;
+    // No asset directory so this is not a composer-based install
+    if (!$assetDir) {
+      return [];
+    }
+    $errorMsg = NULL;
+    $versionFile = $assetDir . '/js/version.json';
+    if (!file_exists($versionFile)) {
+      $errorMsg = ts('Assets missing from public web directory.');
+    }
+    else {
+      $currentVersion = CRM_Utils_System::version();
+      $assetVersion = json_decode(file_get_contents($versionFile), TRUE);
+      if ($assetVersion !== $currentVersion) {
+        $errorMsg = ts('Wrong version of assets in public web directory: expected %1 but found %2.', [1 => $currentVersion, 2 => $assetVersion]);
+      }
+    }
+    if ($errorMsg) {
+      $checks[] = new CRM_Utils_Check_Message(
+        __FUNCTION__,
+        $errorMsg .
+        '<br /><a target="_blank" href="https://docs.civicrm.org/sysadmin/en/latest/upgrade/drupal8/#ensure-that-civicrm-assets-are-up-to-date">' .
```

Comment:
Docs link has already gone stale - it's now `#additional-cleanup`.
Do you want to update that and then I'd merge this.

Reply:
Also should probably use docURL2, like at https://github.com/civicrm/civicrm-core/blob/5c4ecf855000ecbdfc82782f01c3cdadf76212c4/CRM/Utils/Check/Component/Env.php#L123

Reply:
@demeritcowboy updated, thanks for the suggestion